src/main.c

```diff
@@ -371,73 +406,77 @@ static void sia_main(void) {
 // with the simplest command, signHash.c.
 
 // override point, but nothing more to do
+#ifdef HAVE_BAGL
 void io_seproxyhal_display(const bagl_element_t *element) {
 	io_seproxyhal_display_default((bagl_element_t *)element);
 }
+#endif
 
-unsigned char G_io_seproxyhal_spi_buffer[IO_SEPROXYHAL_BUFFER_SIZE_B];
-
-unsigned char io_event(unsigned char channel __attribute__((unused))) {
-	// can't have more than one tag in the reply, not supported yet.
-	switch (G_io_seproxyhal_spi_buffer[0]) {
-	case SEPROXYHAL_TAG_FINGER_EVENT:
-		UX_FINGER_EVENT(G_io_seproxyhal_spi_buffer);
-		break;
-
-	case SEPROXYHAL_TAG_BUTTON_PUSH_EVENT:
-		UX_BUTTON_PUSH_EVENT(G_io_seproxyhal_spi_buffer);
-		break;
-
-	case SEPROXYHAL_TAG_STATUS_EVENT:
-		if (G_io_apdu_media == IO_APDU_MEDIA_USB_HID &&
-			!(U4BE(G_io_seproxyhal_spi_buffer, 3) &
-			  SEPROXYHAL_TAG_STATUS_EVENT_FLAG_USB_POWERED)) {
-			THROW(EXCEPTION_IO_RESET);
-		}
-		UX_DEFAULT_EVENT();
-		break;
-
-	case SEPROXYHAL_TAG_DISPLAY_PROCESSED_EVENT:
-		UX_DISPLAYED_EVENT({});
-		break;
-
-	case SEPROXYHAL_TAG_TICKER_EVENT:
-		UX_TICKER_EVENT(G_io_seproxyhal_spi_buffer, {});
-		break;
-
-	default:
-		UX_DEFAULT_EVENT();
-		break;
-	}
+uint8_t io_event(uint8_t channel) {
+    (void)channel;
+
+    switch (G_io_seproxyhal_spi_buffer[0]) {
+        case SEPROXYHAL_TAG_BUTTON_PUSH_EVENT:
+#ifdef HAVE_BAGL
+            UX_BUTTON_PUSH_EVENT(G_io_seproxyhal_spi_buffer);
+#endif  // HAVE_BAGL
+            break;
+        case SEPROXYHAL_TAG_STATUS_EVENT:
+            if (G_io_apdu_media == IO_APDU_MEDIA_USB_HID &&  //
+                !(U4BE(G_io_seproxyhal_spi_buffer, 3) &      //
+                  SEPROXYHAL_TAG_STATUS_EVENT_FLAG_USB_POWERED)) {
+                THROW(EXCEPTION_IO_RESET);
+            }
+            /* fallthrough */
+        case SEPROXYHAL_TAG_DISPLAY_PROCESSED_EVENT:
+#ifdef HAVE_BAGL
+            UX_DISPLAYED_EVENT({});
+#endif  // HAVE_BAGL
+#ifdef HAVE_NBGL
```

Comment:
I thought this was redundant, but I see now that it's `NBGL`, not `BAGL`. Nice of them to make them look so similar...

all of these `ifdef`s bum me out, but not much we can do about it I guess